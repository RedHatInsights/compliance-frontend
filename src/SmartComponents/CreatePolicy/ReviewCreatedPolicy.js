import React from 'react';
import {
    Text,
    TextVariants,
    TextContent,
    TextList,
    TextListVariants,
    TextListItem,
    TextListItemVariants
} from '@patternfly/react-core';
import { formValueSelector } from 'redux-form';
import { connect } from 'react-redux';
import gql from 'graphql-tag';
import propTypes from 'prop-types';
import { Spinner } from '@redhat-cloud-services/frontend-components';
import { useQuery } from '@apollo/react-hooks';

const REVIEW = gql`
query review($benchmarkId: String!) {
    benchmark(id: $benchmarkId) {
        title
        refId
        version
    }
}
`;

const ReviewCreatedPolicy = ({ benchmarkId, name, refId, systemsCount, rulesCount, complianceThreshold, parentProfileName }) => {
    const { data, error, loading } = useQuery(REVIEW, { variables: { benchmarkId } });

    if (error) { return error; }

    if (loading) { return <Spinner/>; }

    const benchmark = data.benchmark;

    return (
        <TextContent>
            <Text component={TextVariants.h1}>
                Review
            </Text>
            <Text component={TextVariants.h4}>
                Review your policy before finishing.
            </Text>
            <TextList component={TextListVariants.dl}>
                <TextListItem component={TextListItemVariants.dt}>Operating system</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>
                    { benchmark.refId.split('xccdf_org.ssgproject.content_benchmark_')[1].replace('-', ' ') }
                </TextListItem>
                <TextListItem component={TextListItemVariants.dt}>Security guide</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>
                    {` ${benchmark.title} - ${benchmark.version}`}
                </TextListItem>
                <TextListItem component={TextListItemVariants.dt}>Policy type</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>{ parentProfileName }</TextListItem>
                <TextListItem component={TextListItemVariants.dt}>Policy name</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>{ name }</TextListItem>
                <TextListItem component={TextListItemVariants.dt}>Reference ID</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>{ refId }</TextListItem>
                <TextListItem component={TextListItemVariants.dt}>Compliance threshold</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>{ complianceThreshold }%</TextListItem>
                <TextListItem component={TextListItemVariants.dt}>No. of rules</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>{ rulesCount }</TextListItem>
                <TextListItem component={TextListItemVariants.dt}>No. of systems</TextListItem>
                <TextListItem component={TextListItemVariants.dd}>{ systemsCount }</TextListItem>
            </TextList>
        </TextContent>
    );
};

ReviewCreatedPolicy.propTypes = {
    benchmarkId: propTypes.string,
    refId: propTypes.string,
    name: propTypes.string,
    systemsCount: propTypes.number,
    rulesCount: propTypes.number,
    complianceThreshold: propTypes.number,
    parentProfileName: propTypes.string
};

const selector = formValueSelector('policyForm');

export default connect(
    state => ({
        benchmarkId: selector(state, 'benchmark'),
        refId: selector(state, 'refId'),
        name: selector(state, 'name'),
        systemsCount: selector(state, 'systems').length,
        complianceThreshold: parseFloat(selector(state, 'complianceThreshold')) || 100.0,
        parentProfileName: JSON.parse(selector(state, 'profile')).name,
        rulesCount: selector(state, 'selectedRuleRefIds').length
    })
)(ReviewCreatedPolicy);
