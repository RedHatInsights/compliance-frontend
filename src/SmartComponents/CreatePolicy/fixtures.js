export const policyFormValues = {
    benchmark: 'a5e7f1ea-e63c-40be-a17a-c2a247c11e10',
    description: 'This profile demonstrates compliance against the U.S. Government Commercial Cloud Services (C2S)',
    name: 'C2S for Red Hat Enterprise Linux 6',
    profile: JSON.stringify({
        complianceThreshold: 100,
        description: 'This profile demonstrates compliance against the U.S. Government Commercial Cloud Services (C2S)',
        id: '197eb783-7bca-45c5-978d-c1e89b0118da',
        name: 'C2S for Red Hat Enterprise Linux 6',
        refId: 'xccdf_org.ssgproject.content_profile_C2S'
    }),
    refId: 'xccdf_org.ssgproject.content_profile_C2S',
    systems: []
};
