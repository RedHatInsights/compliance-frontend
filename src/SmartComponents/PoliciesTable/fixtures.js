export const policies = {
    edges: [
        {
            node: {
                id: 'b71376fd-015e-4209-99af-4543e82e5dc5',
                name: 'United States Government Configuration Baseline23',
                refId: 'xccdf_org.ssgproject.content_profile_ospp23',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '719999b6-d230-4ba5-8dba-7ab3dc6561e0',
                name: 'PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 73',
                refId: 'xccdf_org.ssgproject.content_profile_pci-dss3',
                complianceThreshold: 67,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: 'dae0487d-3201-4ee0-af5f-b94cde2af818',
                name: 'United States Government Configuration Baseline2',
                refId: 'xccdf_org.ssgproject.content_profile_ospp2',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '20a9d997-62a6-40cc-a5f3-19d466eb975e',
                name: 'C2S for Red Hat Enterprise Linux 7',
                refId: 'xccdf_org.ssgproject.content_profile_C2S',
                complianceThreshold: 69.5,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '6d345bd2-d597-4df8-9bcf-71c41155b42c',
                name: 'Criminal Justice Information Services (CJIS) Security Policy',
                refId: 'xccdf_org.ssgproject.content_profile_cjis',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: 'c8e15347-9c2b-495d-8e54-503c2f9582b6',
                name: 'Unclassified Information in Non-federal Information Systems and Organizations (NIST 800-171)',
                refId: 'xccdf_org.ssgproject.content_profile_nist-800-171-cui',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '3c4823a1-2c16-46ae-b2fe-0cebf5a03931',
                name: 'PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 7',
                refId: 'xccdf_org.ssgproject.content_profile_pci-dss',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: 'f7b7977a-403b-4cd1-ab90-20b6f9a5a359',
                name: 'Red Hat Corporate Profile for Certified Cloud Providers (RH CCP)',
                refId: 'xccdf_org.ssgproject.content_profile_rht-ccp',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '36abc364-6dc3-4e35-94f4-d10fa77e866e',
                name: 'Health Insurance Portability and Accountability Act (HIPAA)',
                refId: 'xccdf_org.ssgproject.content_profile_hipaa',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: 'd35c8aad-8fc8-49e8-bff0-4d9d3dc8f220',
                name: 'United States Government Configuration Baseline',
                refId: 'xccdf_org.ssgproject.content_profile_ospp',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '4c27fe09-9a7f-437c-b38b-e42272d9ccf0',
                name: 'Standard System Security Profile for Red Hat Enterprise Linux 7',
                refId: 'xccdf_org.ssgproject.content_profile_standard',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '9b034440-e3dd-4c19-8f2c-ca75e813d57d',
                name: 'DISA STIG for Red Hat Enterprise Linux 7',
                refId: 'xccdf_org.ssgproject.content_profile_stig-rhel7-disa',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        },
        {
            node: {
                id: '19921ca4-8526-4651-8876-3c8587e8e125',
                name: 'PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 72',
                refId: 'xccdf_org.ssgproject.content_profile_pci-dss2',
                complianceThreshold: 100,
                businessObjective: null,
                __typename: 'Profile'
            },
            __typename: 'ProfileEdge'
        }
    ],
    __typename: 'ProfileConnection'
}
