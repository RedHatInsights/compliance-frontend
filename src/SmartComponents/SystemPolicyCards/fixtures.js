export const fixturesSystemReports = {
  data: [
    {
      id: '0de744d1-b00d-4b6c-a524-de0eb94dbe97',
      title:
        'VPP - Protection Profile for Virtualization v. 1.0 for Red Hat Virtualization',
      description:
        'This compliance profile reflects the core set of security\nrelated configuration settings for deployment of Red Hat Enterprise\nLinux Hypervisor (RHELH) 7.x into U.S. Defense, Intelligence, and Civilian agencies.\nDevelopment partners and sponsors include the U.S. National Institute\nof Standards and Technology (NIST), U.S. Department of Defense,\nthe National Security Agency, and Red Hat.\n\nThis baseline implements configuration requirements from the following\nsources:\n\n- Committee on National Security Systems Instruction No. 1253 (CNSSI 1253)\n- NIST 800-53 control selections for MODERATE impact systems (NIST 800-53)\n- U.S. Government Configuration Baseline (USGCB)\n- NIAP Protection Profile for Virtualization v1.0 (VPP v1.0)\n\nFor any differing configuration requirements, e.g. password lengths, the stricter\nsecurity setting was chosen. Security Requirement Traceability Guides (RTMs) and\nsample System Security Configuration Guides are provided via the\nscap-security-guide-docs package.\n\nThis profile reflects U.S. Government consensus content and is developed through\nthe ComplianceAsCode project, championed by the National\nSecurity Agency. Except for differences in formatting to accommodate\npublishing processes, this profile mirrors ComplianceAsCode\ncontent as minor divergences, such as bugfixes, work through the\nconsensus and release processes.',
      business_objective: null,
      compliance_threshold: 100.0,
      type: 'report',
      os_major_version: 7,
      profile_title:
        'VPP - Protection Profile for Virtualization v. 1.0 for Red Hat Virtualization',
      ref_id: 'xccdf_org.ssgproject.content_profile_rhelh-vpp',
      all_systems_exposed: false,
      percent_compliant: 0,
      compliant_system_count: 0,
      unsupported_system_count: 0,
      reported_system_count: 1,
    },
    {
      id: '2187dfbe-16b2-411d-8f57-bb5cc9f8449f',
      title: 'ANSSI-BP-028 (enhanced)',
      description:
        "This profile contains configurations that align to ANSSI-BP-028 v2.0 at the enhanced hardening level.\n\nANSSI is the French National Information Security Agency, and stands for Agence nationale de la sécurité des systèmes d'information.\nANSSI-BP-028 is a configuration recommendation for GNU/Linux systems.\n\nA copy of the ANSSI-BP-028 can be found at the ANSSI website:\nhttps://www.ssi.gouv.fr/administration/guide/recommandations-de-securite-relatives-a-un-systeme-gnulinux/\n\nAn English version of the ANSSI-BP-028 can also be found at the ANSSI website:\nhttps://cyber.gouv.fr/publications/configuration-recommendations-gnulinux-system",
      business_objective: null,
      compliance_threshold: 100.0,
      type: 'report',
      os_major_version: 7,
      profile_title: 'ANSSI-BP-028 (enhanced)',
      ref_id: 'xccdf_org.ssgproject.content_profile_anssi_nt28_enhanced',
      all_systems_exposed: false,
      percent_compliant: 0,
      compliant_system_count: 0,
      unsupported_system_count: 0,
      reported_system_count: 1,
    },
    {
      id: '3d55732c-1170-4f8e-83ed-b56760f9301d',
      title: 'Standard System Security Profile for Red Hat Enterprise Linux 7',
      description:
        "This profile contains rules to ensure standard security baseline\nof a Red Hat Enterprise Linux 7 system. Regardless of your system's workload\nall of these checks should pass.",
      business_objective: null,
      compliance_threshold: 100.0,
      type: 'report',
      os_major_version: 7,
      profile_title:
        'Standard System Security Profile for Red Hat Enterprise Linux 7',
      ref_id: 'xccdf_org.ssgproject.content_profile_standard',
      all_systems_exposed: false,
      percent_compliant: 0,
      compliant_system_count: 0,
      unsupported_system_count: 0,
      reported_system_count: 1,
    },
  ],
};

export const fixturesReportTestResults = {
  data: [
    {
      id: '14d74821-1918-4499-a77e-734cba7fb549',
      end_time: '2024-09-11T09:23:03.000Z',
      failed_rule_count: 108,
      supported: true,
      score: 64.703888,
      type: 'test_result',
      display_name: 'rhiqe.ptiw11.test',
      groups: [],
      tags: [
        { key: 'pnRsctS', value: 'qqeixff', namespace: 'pasjBk' },
        { key: 'yXgnLy', value: 'KpNJKvv', namespace: 'WkBkgsDI' },
        { key: 'FVaACO', value: 'WdzcBw', namespace: 'WywAdxiL' },
        { key: 'LAkbZeyeJK', value: 'ZyPXMw', namespace: 'gJRWgGQm' },
        { key: 'nKcxvk', value: 'mBorSc', namespace: 'huIorTGAhs' },
      ],
      os_major_version: 7,
      os_minor_version: 8,
      compliant: false,
      system_id: '70d288a3-3691-47dc-a225-b87aa014a511',
      security_guide_version: '0.1.46',
    },
  ],
};
