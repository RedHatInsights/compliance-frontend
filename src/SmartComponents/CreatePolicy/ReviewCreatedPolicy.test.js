import ReviewCreatedPolicy from './ReviewCreatedPolicy.js';
import configureStore from 'redux-mock-store';
import { policyFormValues } from './fixtures.js';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { useQuery } from '@apollo/react-hooks';

const mockStore = configureStore();
jest.mock('@apollo/react-hooks');
jest.mock('../CreatePolicy/EditPolicyRules', () => {
    return <p>Systems table</p>;
});

describe('ReviewCreatedPolicy', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({ form: { policyForm: { values: policyFormValues } } });
    });

    it('expect to render without error', () => {
        useQuery.mockImplementation(() => ({ data: {
            benchmark: {
                title: 'SCAP security guide for RHEL7',
                refId: 'xccdf_org.ssgproject.content_benchmark_RHEL-7',
                version: '0.1.40'  }
        }, error: false, loading: false }));
        component = renderer.create(
            <Provider store={store}>
                <ReviewCreatedPolicy />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render a spinner while loading', () => {
        useQuery.mockImplementation(() => ({ data: {}, error: false, loading: true }));
        component = renderer.create(
            <Provider store={store}>
                <ReviewCreatedPolicy />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
