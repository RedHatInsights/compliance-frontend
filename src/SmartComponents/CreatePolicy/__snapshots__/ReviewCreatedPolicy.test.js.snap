// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCreatedPolicy expect to render without error 1`] = `
<div
  className="pf-c-content"
>
  <h1
    className=""
    data-pf-content={true}
  >
    Review
  </h1>
  <h4
    className=""
    data-pf-content={true}
  >
    Review your policy before finishing.
  </h4>
  <dl
    className=""
    data-pf-content={true}
  >
    <dt
      className=""
      data-pf-content={true}
    >
      Operating system
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      RHEL 7
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      Security guide
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
       SCAP security guide for RHEL7 - 0.1.40
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      Policy type
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      C2S for Red Hat Enterprise Linux 6
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      Policy name
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      C2S for Red Hat Enterprise Linux 6
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      Reference ID
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      xccdf_org.ssgproject.content_profile_C2S
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      Compliance threshold
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      100
      %
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      No. of rules
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      1
    </dd>
    <dt
      className=""
      data-pf-content={true}
    >
      No. of systems
    </dt>
    <dd
      className=""
      data-pf-content={true}
    >
      0
    </dd>
  </dl>
</div>
`;

exports[`ReviewCreatedPolicy should render a spinner while loading 1`] = `
<div
  className="ins-c-spinner"
  role="status"
>
  <span
    className="pf-u-screen-reader"
  >
    Loading...
  </span>
</div>
`;
